### PR TITLE
vdirsyncer: 0.17.0a3 -> 2018-08-05

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

Use the git version of vdirsyncer which works with the latest `shippai` version.  See #52780.

cc @jgeerds @FRidh 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/vdirsyncer/default.nix [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
-  pythonPackages = python3Packages;
-  version = "0.17.0a3";
+python3Packages.buildPythonApplication rec {
+  version = "2018-08-05";
   pname = "vdirsyncer";
-  name = pname + "-" + version;
-  src = pythonPackages.fetchPypi {
-    inherit pname version;
-    sha256 = "1n7izfa5x9mh0b4zp20gd8qxfcca5wpjh834bsbi5pk6zam5pfdy";
+  name = "${pname}-${version}";
```
Comment (worldofpeace) [inline pkgs/tools/misc/vdirsyncer/default.nix]:
```suggestion
```
This is done automatically.

File: pkgs/tools/misc/vdirsyncer/default.nix [hunk 2]
```diff
@@ -1,27 +1,28 @@
-{ stdenv, python3Packages, fetchpatch, glibcLocales, rustPlatform, pkgconfig, openssl, Security }:
+{ stdenv, python3Packages, fetchFromGitHub, fetchpatch, glibcLocales, rustPlatform, pkgconfig, openssl, Security }:
 
 # Packaging documentation at:
 # https://github.com/untitaker/vdirsyncer/blob/master/docs/packaging.rst
-let
-  pythonPackages = python3Packages;
-  version = "0.17.0a3";
+python3Packages.buildPythonApplication rec {
+  version = "2018-08-05";
```
Comment (worldofpeace) [inline pkgs/tools/misc/vdirsyncer/default.nix]:
```suggestion
  version = "unstable-2018-08-05";
```

When version is not a release but a commit from a repository we use this format.

**Comment by worldofpeace** [comment]
@gebner Can you squash https://github.com/NixOS/nixpkgs/pull/53040/commits/6d393084512911d175ea593039d53c134908f88a and https://github.com/NixOS/nixpkgs/pull/53040/commits/a84a80fb4fe9d4aa5e3097c70c74cd129f0af91d?

Also correct the msg.

File: pkgs/tools/misc/vdirsyncer/default.nix (hunk 1 above)
Comment (gebner) [reply pkgs/tools/misc/vdirsyncer/default.nix]: Unfortunately only at the level of the derivation's attributes.  However here the `name` attribute is directly used a few lines further down.

File: pkgs/tools/misc/vdirsyncer/default.nix (hunk 2 above)
Comment (gebner) [reply pkgs/tools/misc/vdirsyncer/default.nix]: Thanks, forgot about that.

**Comment by gebner** [comment]
Done and thanks!

**Pull request merged**